spark/src/main/java/org/opensearch/sql/spark/cluster/FlintStreamingJobCleanerTask.java

```diff
@@ -0,0 +1,134 @@
+                // Datasource disabled.
+                LOGGER.info("Datasource is deleted for autoRefreshIndex: {}", autoRefreshIndex);
+                dropAutoRefreshIndex(autoRefreshIndex, flintIndexMetadata, datasourceName);
+              }
+            } catch (Exception exception) {
+              LOGGER.error(
+                  "Failed to alter/cancel index {}: {}",
+                  autoRefreshIndex,
+                  exception.getMessage(),
+                  exception);
+              Metrics.getInstance()
+                  .getNumericalMetric(MetricName.STREAMING_JOB_CLEANER_TASK_FAILURE_COUNT)
+                  .increment();
+            }
+          });
+    } catch (Throwable error) {
+      LOGGER.error("Error while running the streaming job cleaner task: {}", error.getMessage());
+      Metrics.getInstance()
+          .getNumericalMetric(MetricName.STREAMING_JOB_CLEANER_TASK_FAILURE_COUNT)
+          .increment();
+    } finally {
+      isRunning.set(false);
+    }
+  }
+
+  private void dropAutoRefreshIndex(
+      String autoRefreshIndex, FlintIndexMetadata flintIndexMetadata, String datasourceName) {
+    // When the datasource is deleted. Possibly Replace with VACUUM Operation.
+    LOGGER.info("Attempting to drop auto refresh index: {}", autoRefreshIndex);
+    FlintIndexOpDrop flintIndexOpDrop =
+        new FlintIndexOpDrop(stateStore, datasourceName, emrServerlessClientFactory.getClient());
+    flintIndexOpDrop.apply(flintIndexMetadata);
+    LOGGER.info("Successfully dropped index: {}", autoRefreshIndex);
+  }
+
+  private void alterAutoRefreshIndex(
+      String autoRefreshIndex, FlintIndexMetadata flintIndexMetadata, String datasourceName) {
+    LOGGER.info("Attempting to alter index: {}", autoRefreshIndex);
+    FlintIndexOptions flintIndexOptions = new FlintIndexOptions();
+    flintIndexOptions.setOption(FlintIndexOptions.AUTO_REFRESH, "false");
+    FlintIndexOpAlter flintIndexOpAlter =
+        new FlintIndexOpAlter(
+            flintIndexOptions,
+            stateStore,
+            datasourceName,
+            emrServerlessClientFactory.getClient(),
+            flintIndexMetadataService);
+    flintIndexOpAlter.apply(flintIndexMetadata);
+    LOGGER.info("Successfully altered index: {}", autoRefreshIndex);
+  }
+
+  private String getDataSourceName(FlintIndexMetadata flintIndexMetadata) {
+    String kind = flintIndexMetadata.getKind();
+    switch (kind) {
+      case "mv":
+        return flintIndexMetadata.getName().split("\\.")[0];
+      case "skipping":
+      case "covering":
+        return flintIndexMetadata.getSource().split("\\.")[0];
+      default:
+        throw new IllegalArgumentException("Unknown flint index kind.");
+    }
+  }
+
+  private Map<String, FlintIndexMetadata> getAllAutoRefreshIndices() {
+    Map<String, FlintIndexMetadata> flintIndexMetadataHashMap =
+        flintIndexMetadataService.getFlintIndexMetadata("flint_*");
```

Comment:
If any non-Flint index matched, will it block all the logic after this? Or it will be ignored?

Reply:
It will result in failure. As the metadata doesn't contain any of the required information.

Reply:
We may need to ignore those with matched name though created by user?

Reply:
How do we identify from the meta information?

Reply:
On Spark side, I recall it's identified by deserialization error. In this case, maybe JSON path "_meta.options" is valid or not? We can do this later if not minor changes.